### ISSUE TITLE
508 Critical: Prevent submission of "select" default value

### ISSUE THREAD

**Issue by mfloyd-iat** [department-of-veterans-affairs/vets-design-system-documentation#5205]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

The implementation of AskVA on VA.gov raised some accessibility findings during an audit.  This issue is related to using VaSelect on the AskVA form

- On the Ask VA page → Ask a new question link → Continue button flow, the placeholder “- Select -” in the combo box is incorrectly treated as a valid value instead of an instruction.
- Example: In the *“Select the category that best describes your question (Required)” combo box, “- Select -” appears as an actual option.
- When a user selects Health care, clicks Continue, then clicks Back and selects “- Select -”, the system still allows the user to submit and advance to the next page.
- This violates WCAG 3.3.2 (Labels or Instructions) because the placeholder does not function as an instruction and validation is bypassed.

## What I expected to happen

- Once the user has selected a value, "--Select--" should not be available as an option in the dropdown

## Reproducing

The replication steps were completed on Chrome on a Mac - but the nature of this issue doesn't seem to be platform specific.

**URL(s) where this bug occurs**

- Start of the form:  https://staging.va.gov/contact-us/ask-va/introduction
- Bug occurs: https://staging.va.gov/contact-us/ask-va/category-topic-1

**Steps to reproduce:**

1. Navigate to the Ask VA page. [AskVA staging](https://staging.va.gov/contact-us/ask-va/introduction)
2. Click Ask a new question.
3. Locate the combo box labeled *“Select the category that best describes your question (Required)”.
4. Observe that “-- Select --” is listed as a value.
5. Choose Health care and click Continue.
6. Click the Back button.
7. Select “- Select -” from the dropdown.
8. Click Continue.
9. Notice that the system allows submission to the next page, even though a valid option was not selected.

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [x] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

This was found during an accessibility audit of AskVA's form.  It is listed on the report as Critical

### Details

The AskVA form is using the VASelect to display a list of categories to submit an inquiry.  The select is populated using an API call.  This is the original issue that was raised by accessibility testing:

[1935 - 508 Critical: Prevent submission of "select" default value](https://github.com/department-of-veterans-affairs/ask-va/issues/1935)

Defect 3 | Critical | Created:
WCAG 3.3.2 Labels or Instructions | Labels or instructions are provided when content requires user input
Ask VA page -> Ask a new question link -> Continue button -> Continue button
Throughout the product, the instruction “- Select -” is provided as a value of the list combo box. This incorrectly conveys that this is a value, rather than an instruction. For example, on the “Select the category that best describes your question (*Required)” combo box the “-Select-” is list as a value. In addition, when chose “Health care” from the dropdown panel and click “Continue” button and then click “Back” button and then chose “-Select-“, user is able to submit successfully to the next page. Please see attached screenshot.

<img width="858" height="521" alt="Image" src="https://github.com/user-attachments/assets/134feb00-d507-4ab8-9513-d33a86c709d2" />

**Comment by it-harrison** [comment]
@mfloyd-iat Hi Matt, just taking a look at this now. It seems there is a bug in the app: when a user selects the "Select" option there is [some code](https://github.com/department-of-veterans-affairs/vets-website/blob/4c5e5a28bd472a3f48e588df0a55d4c6fb7905b7/src/applications/ask-va/containers/CategorySelectPage.jsx#L46) that is executed that results in an error; because the error is thrown, the form data from the previous selection is not cleared and the user can proceed even though it seems that this should not be possible. And you can see the error in the console, too:

<img width="718" height="135" alt="Image" src="https://github.com/user-attachments/assets/eabf708c-3c2a-4cc8-af2c-787929b7e098" />
